mainloop/game-loop-designs/02-fuggetlen_main_loopok/game_over.cc

```diff
@@ -0,0 +1,104 @@
+#include <SFML/Graphics.hpp>
+#include <cstdint>
+#include <cstdlib>  // for EXIT_FAILURE
+#include <cwctype>
+#include <vector>
+#include <functional>
+
+class Button : public sf::Drawable {
+  public:
+    Button(sf::Font const& font, std::string label, sf::Vector2f const& pos)
+      : txt{sf::String::fromUtf8(label.begin(), label.end()), font, 50}, result{label} {
+      txt.setPosition(pos);
+      txt.setFillColor(sf::Color::Green);
+      sf::FloatRect txtBounds = txt.getGlobalBounds();
+      box.setPosition(sf::Vector2f{pos.x - 10.0f, pos.y - 10.0f});
+      box.setSize(sf::Vector2f{txtBounds.width + 30.0f, 80.0f});
+      box.setOutlineColor(sf::Color::Green);
+      box.setOutlineThickness(2.0f);
+      box.setFillColor(sf::Color::Black);
+    }
+
+    std::string get_result() const {
+      return result;
+    }
+
+    virtual void draw(sf::RenderTarget& target, sf::RenderStates states) const {
+      if (selected) {
+        target.draw(box);
+      }
+      target.draw(txt);
+    }
+
+    bool selected = false;
+  private:
+    sf::RectangleShape box;
+    sf::Text txt;
+    std::string result;
+};
+
+bool game_over(sf::RenderWindow &window) {
```

Comment:
Please add a documentation comment, that describes what the return value means. Something like:

// Returns true if the player requested another round.